sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
```

Comment:
I guess this won't work if someone wants `heaviside(x, S.NaN)`?

Why not `elif arg.is_zero and H0 is not None`?   (and drop the `H0 == None:` bit above)?